lib/dry/transaction/step.rb

```diff
@@ -38,39 +39,62 @@ def with(operation: UNDEFINED, call_args: UNDEFINED)
           operation_name,
           new_operation,
           options,
-          new_call_args,
+          new_call_args
         )
       end
 
-      def call(input)
+      def call(input, next_step = NOOP)
         args = [input] + Array(call_args)
+        continue = once(next_step)
+
+        with_broadcast(args, continue) do
+          step_adapter.call(self, *args, &continue)
+        end
+      end
+
+      def once(next_step)
```

Comment:
I know this is looking _smarter_ then needed but distinguishing `around` steps from others would require changing the API of step adapters but I'm not sure if we want to do it this time, although we could.

Reply:
How do you think things would look if you did change the step adapter API? If we're going to change it, now is _kind of_ the time to do it, while we're still pre-1.0 and discovering (with the introduction of this step) the various different things we might want to be doing.

Reply:
So AFAICT the reason for this is that the step adapter may call the `&continue` proc more than once? Would you mind pointing out where/how this might happen?

Reply:
re API, I need to play with it a bit, I'll come up with an example later. Maybe adding something like `yields?` to the adapter would be enough
re `once` `continue` will be called _once_ on every regular step and called _twice_ on an around step: the first time in `yield` of the step definition and the second on the explicit `continue.(result)` call. TBH, having a second thought I realized this has some drawbacks, for instance, it doesn't allow to halt the execution in an around step and I imagine this could be useful somehow.